js/net_test.ts

```diff
@@ -8,6 +8,37 @@ testPerm({ net: true }, function netListenClose() {
   listener.close();
 });
 
+testPerm({ net: true }, async function netCloseWhileAccept() {
+  const listener = deno.listen("tcp", ":4501");
+  const p = listener.accept();
+  listener.close();
+  let err: Error;
+  try {
+    await p;
+  } catch (e) {
+    err = e;
+  }
+  assert(!!err);
+  assertEqual(err.message, "Listener has been closed");
+});
+
+testPerm({ net: true }, async function netConcurrentAccept() {
+  const listener = deno.listen("tcp", ":4502");
+  let err: Error;
+  // Consume this accept error
+  // (since it would still be waiting when listener.close is called)
+  listener.accept().catch(() => {});
```

Comment:
It would be `Other` for now (I could not find a better `ErrorKind` under `std::io::Error`. I guess we could choose a reasonable type for this later...)